system/forcefield.py

```diff
@@ -264,19 +265,26 @@ def add_param(p, p_group):
                     all_k_idxs.append(k_idx)
                     all_phase_idxs.append(phase_idx)
                     all_period_idxs.append(period_idx)
-
-                for m in matches:
+                
+                # remove the redundant matches and use unique pair for trefoil
+                new_matches = []
+                
+                for element in matches:
+                    tmp=[element[0], element[2], element[3]]
```

Comment:
Style nitpick

Add a space before and after the =

Reply:
Got it!